### ISSUE TITLE
Youtube comment caching issue

### ISSUE THREAD

**Issue by atomheartother** [twstokes/herpderp#12]
This is a *really* strange bug, I think it popped up with Youtube's new layout/ client code. It's easy to reproduce, too!

Using either FF or Chrome, go to any youtube video and click it. Let comments get herped, then *without reloading*, go back to the previous page, the page you clicked the video from. Then re-click on the video. Go look at the comments... and poof:

![screenshot from 2017-12-15 01-26-06](https://user-images.githubusercontent.com/10134879/34022405-fbd46988-e136-11e7-961c-3d165f579451.png)

There's our herped comments... and then the original comment, appended to the herp string. I've tested, and this is indeed the original comment there, and that's where things get weird: if we click on the herped comment, a *completely different* comment appears! One which doesn't belong at this spot:

![screenshot from 2017-12-15 01-29-00](https://user-images.githubusercontent.com/10134879/34022476-5d660f80-e137-11e7-9259-c2bfb4d6016b.png)

Where do these comments come from? If we disable herpderp and look at the comments, we can find the mystery comments at the bottom of the page, in inverse order:

![screenshot from 2017-12-15 01-32-13](https://user-images.githubusercontent.com/10134879/34022545-d393a10e-e137-11e7-9394-5b69412913bc.png)

So, to sum up, after clicking on the same video twice:
- The original comment remains visible, appended to a herped string
- Clicking on it reveals a completely different comment, which comes from the other end of the comments section

As far as I can tell, this is due to Youtube caching DOM in some way to avoid having to request everything again. Why herpderp's changes appear to be applied in reverse order to the DOM though, that I have no clue.

**Comment by twstokes** [comment]
Good catch! I've noticed this as well but have lacked the time to really trace it. I made some progress on the fix but didn't nail it, and will update this issue once I figure it out. I think I was fighting the shadow DOM to inject content.

**Comment by atomheartother** [comment]
Feel free to share your progress here if you haven't figured it out yet and want some input, I've been looking into it a bit myself, but all the fixes I've come up with so far have been pretty dirty... Good to know you'd caught it though!

**Comment by twstokes** [comment]
Will do! I may have some time in the coming weeks to revisit.

**Comment by carbolymer** [comment]
I have the same issue. Any progress on that?

**Comment by twstokes** [comment]
I will try to revisit the problem this week. It's been driving me crazy.

**Comment by twstokes** [comment]
I've uploaded version 1.6.5 tonight thanks to the awesome help of @atomheartother. It should be available soon and helps avoid this problem.
